### ISSUE TITLE
Publish to npm?

### ISSUE THREAD

**Issue by Paul-Reed** [bartbutenaers/node-red-contrib-acme-client#2]
I've been using this on 2 linux machines now for what seems like years, Oh hang on, yes, it has been years !!!
It's worked great, and a shame not to share to the community, are you intending to npm publish it?

**Comment by bartbutenaers** [comment]
Hi @Paul-Reed,
Well it was due to your Cloudflare issues with this node - at the time being - that I have never published it.

Todo's for me:
- The readme is way too complicated.  All the introduction stuff needs to be moved to wiki pages.
- I need to make very clear on the readme that only duckdns has been tested unfortunately
- The password generation should be removed, since it has nothing to do with Letsencrypt.
- The keystore stuff also has to be moved to a keystore node.  This node should send the new certificate in the output message, and the next node in the flow should import it into the keystore.

**Comment by Paul-Reed** [comment]
> Well it was due to your Cloudflare issues with this node - at the time being - that I have never published it.

The issue as I recall was that an error message was in the log about not finding the TXT record -  https://discourse.nodered.org/t/acme/27043/41 although certificate renewals worked OK, and have done so for the past 18 months with no problems whatsoever (does that mean that it's also tested OK on Cloudflare?)

The node makes the whole process so easy, and IMO it's one of your most useful nodes for us beginners.

**Comment by bartbutenaers** [comment]
Well my node creates the TXT record, so I assume this node doesn't work entirely correct then for cloudflare.

Some extra todo's:
1. Do you remember [this ](https://discourse.nodered.org/t/lets-encrypt-root-certificate-change/45866/6?u=bartbutenaers)discussion?  It is still on my todo list, but I haven't checked yet what I need to do for it...

2. I have fixed some time ago a problem a friend of mine had with this node.  It wasn't on Github yet, so I have now created a [pull-request](https://github.com/bartbutenaers/node-red-contrib-letsencrypt/pull/3).  Would be nice if you could have a look, and see it is related to the previous point...

And I said above that I would like to split this into two nodes: one node that requests a new certificate from Letsencrypt, and another keypair node.  Because this node does too much in my opinion.  On the other hand it is very convenient for a user that a single node does everything.   Not sure at the moment what to do with that...

**Comment by bartbutenaers** [comment]
And of course I need to update the dependency versions, to get all the fixes...

**Comment by Paul-Reed** [comment]
> Would be nice if you could have a look, and see it is related to the previous point

I don't think its connected.

Maybe update the dependencies & your PR, and we can recheck. I have 3 servers, 1 using your node, and 2 using certbot.
All using cloudflare.
We can then check for errors & recheck  https://check-your-website.server-daten.de to compare.

**Comment by bartbutenaers** [comment]
Ok, deal.  Thanks for the assitance!
I have already rewritten a major part of the readme page, to have a more step by step tutorial.
By doing it step by step, it seems that I also need to move some settings on the config screen to make it more logical to use.
I am going to keep the keystore code inside this node, because it is indeed very simple to use it like it works now.
Will continue tomorrow.

**Comment by bartbutenaers** [comment]
Morning Paul,

I have been thinking about this, but I more and more think that I should create a basic keystore node.
The flow will become a bit longer, but it is much more self explaining and it is much more Node-RED style:

![image](https://user-images.githubusercontent.com/14224149/146133465-8a1ba2ba-e506-4eeb-aa92-798ef99b3fb7.png)

This node currently does WAY too much.
P.S. the function nodes in the screenshot represent the keystore node...

**Comment by Paul-Reed** [comment]
I don't think that many people are using your node at the moment (as it has never been npm published) so it's a good opportunity to make whatever changes that you wish.

I guess breaking the node down into specific flows, may make it easier to troubleshoot problems & future maintain.
